File: analyze_sheet.py
import pandas as pd
import urllib.request
import sys

url = "https://docs.google.com/spreadsheets/d/1l_W_5WhTFwjj88U4IO-FJIb0a8xZK7E9_m97tiZ9GVk/export?format=xlsx"
try:
    urllib.request.urlretrieve(url, "sheet.xlsx")
    xls = pd.ExcelFile("sheet.xlsx")
    for sheet_name in xls.sheet_names:
        if sheet_name in ["Jan", "Feb", "März", "Apr", "Mai", "Jun", "Jul", "Aug", "Sep", "Okt", "Nov", "Dez"]:
            df = pd.read_excel(xls, sheet_name=sheet_name)
            # The summary is usually at the top right or bottom. Let's look for "Übertrag in den nächsten Monat"
            print(f"--- Sheet: {sheet_name} ---")
            for index, row in df.iterrows():
                row_str = " | ".join([str(x) for x in row.values if pd.notna(x)])
                if "SOLL Arbeitszeit" in row_str or "IST Arbeitszeit" in row_str or "Übertrag" in row_str or "ausgezahlt" in row_str:
                    print(row_str)
            print("\n")
except Exception as e:
    print("Error:", e)

File: analyze_sheet.js
const https = require('https');
const fs = require('fs');
const xlsx = require('xlsx');

const url = "https://docs.google.com/spreadsheets/d/1l_W_5WhTFwjj88U4IO-FJIb0a8xZK7E9_m97tiZ9GVk/export?format=xlsx";

https.get(url, (res) => {
    // google drive redirects, so handle 302
    if (res.statusCode === 302 || res.statusCode === 301 || res.statusCode === 303) {
        https.get(res.headers.location, (res2) => {
            const file = fs.createWriteStream("sheet.xlsx");
            res2.pipe(file);
            file.on('finish', () => {
                file.close();
                analyze();
            });
        });
    } else {
        const file = fs.createWriteStream("sheet.xlsx");
        res.pipe(file);
        file.on('finish', () => {
            file.close();
            analyze();
        });
    }
});

function analyze() {
    try {
        const workbook = xlsx.readFile('sheet.xlsx');
        const targetSheets = ["Jan", "Feb", "März", "Apr", "Mai", "Jun", "Jul", "Aug", "Sep", "Okt", "Nov", "Dez"];
        for (const sheetName of workbook.SheetNames) {
            if (targetSheets.includes(sheetName)) {
                console.log(`--- Sheet: ${sheetName} ---`);
                const sheet = workbook.Sheets[sheetName];
                const json = xlsx.utils.sheet_to_json(sheet, {header: 1, raw: false});
                for (const row of json) {
                    const rowStr = row.join(' | ');
                    if (rowStr.includes('SOLL Arbeitszeit') || rowStr.includes('IST Arbeitszeit') || rowStr.includes('Übertrag') || rowStr.includes('ausgezahlt')) {
                        console.log(rowStr);
                    }
                }
                console.log("\n");
            }
        }
    } catch (e) {
        console.error("Error analyzing:", e.message);
    }
}
